### ISSUE TITLE
Silence errors when not on a Kedro project

### ISSUE THREAD

**Issue by astrojuanlu** [kedro-org/vscode-kedro#77]
![image](https://github.com/user-attachments/assets/75fdb1aa-f9bb-4621-90f7-5cba9b1ff7a9)

I get these errors when not on a Kedro project, which are mildly annoying. Not a big deal but wondering if we could make it more subtle.

Or, say, make them visible when we _know_ it's a Kedro project _but_ it seems to be broken for some reason.

**Comment by noklam** [comment]
^ Possible, right now the server initialised when someone click on a Python or YAML server I think. There maybe a more generic pattern we can use to avoid these signals. It's likely there are limitations as this is controlled by VSCode and we cannot write arbitrary code.  We can stop throwing the error but we need to know think about how does user actually know the server is not working otherwise.

**Comment by datajoely** [comment]
> Same issue as https://github.com/kedro-org/vscode-kedro/issues/77 ? We need better way to discover Kedro project https://github.com/kedro-org/vscode-kedro/issues/53, but making is less noisy can be done before this.
> <img width="494" alt="image" src="https://github.com/user-attachments/assets/adca5953-764b-4eb8-9dfe-266d657f503a">
> Ideally I want something like "Mark this as Kedro Project"

@noklam  replying to comment on #137 here - I think for MVP you just need to do something on the VS Cod command palette - probably both "`Disable for this project`" and "`Mark this project as Kedro`" (the second one perhaps needs a fine grained working directory option)

<img width="537" alt="image" src="https://github.com/user-attachments/assets/ac46e0c2-d3f6-4ade-9520-c2fd5af3cc70">

**Comment by datajoely** [comment]
I want to reiterate I think there is a substantial risks users uninstall the extension because it's noisy and distracts you when you're working on something else. I've been working in FastAPI for the last week and I close the pop up every time I open the IDE.

**Comment by noklam** [comment]
> probably both "Disable for this project" and "Mark this project as Kedro" (the second one perhaps needs a fine grained working directory option)

Just confirming my understanding - those warnings are now showing up everytime you open a new VSCode, it should stops showing up once you get the pop up as it will automatically stop restarting the server after a few retry. Would people be bother to run a command and disable `Kedro` specifically for one project? (also, what is the notion of a "project"? I do not want to introduce a new metadata file so it's tricky for the extension to know what exactly is a project, unless user is using a vscode workspace).

The alternative is it's disable by default but need a command to activate the extension (or at some point this get better so the extension can auto-discover the pyproject.toml to locate the `tool.kedro` section).

**Comment by datajoely** [comment]
I think there are two ways in mind:

- Provide a mechanism for the extension to learn and project isn't Kedro. (Reactive)
- Use a popup to say, "this looks like a Kedro project - enable extension?" (Proactive)

**Comment by astrojuanlu** [comment]
As a first, easy step we want to silence the error.

In parallel we'll open a separate issue to have a smarter way of detecting the Kedro project so that we don't even instantiate the LSP if we cannot do so.

**Comment by datajoely** [comment]
<img width="445" alt="image" src="https://github.com/user-attachments/assets/f189364b-2e25-4a49-8b9d-4025c0dbba3e">

this is a scary error message we could handle more gracefully

**Comment by SajidAlamQB** [comment]
Completed in: https://github.com/kedro-org/vscode-kedro/pull/154
